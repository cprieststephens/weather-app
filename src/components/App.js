import { useEffect, useState } from "react";
import LocationDetails from "./LocationDetails";
import SearchForm from "./SearchForm";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import getForecast from "../requests/getForecast";
import "../styles/App.css";

function App() {
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });
  const [selectedDate, setSelectedDate] = useState(0);
  const [searchText, setSearchText] = useState("");
  const [errorMessage, setErrorMessage] = useState("");

  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };

  const handleCitySearch = () => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  };

  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );

  const { city, country } = location;

  useEffect(() => {
    getForecast(
      "Manchester",
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  }, []);

  return (
    <div className="weather-app">
      <div className="weather-app__top">
        <SearchForm
          searchText={searchText}
          setSearchText={setSearchText}
          onSubmit={handleCitySearch}
        />
        <LocationDetails
          city={city}
          country={country}
          errorMessage={errorMessage}
        />
        {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
      </div>
      {!errorMessage && (
        <ForecastSummaries
          forecasts={forecasts}
          onForecastSelect={handleForecastSelect}
        />
      )}
    </div>
  );
}

export default App;
