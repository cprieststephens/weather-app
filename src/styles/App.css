@import url('https://fonts.googleapis.com/css2?family=Raleway&display=swap');

.weather-app {
  display: flex;
  flex-direction: column;
  font-family: 'Raleway', sans-serif;
  background-color: #92d5fd;
  margin: 0 auto;
  border-radius: 9px;
  min-width: 310px;
  padding: 17% 10%;
}

.weather-app__top {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

@media only screen and (max-width: 991px) {
  .weather-app__top {
    background-color: #92d5fd;
    flex-direction: column;
    align-items: center;
    width: 80%;
    position: fixed;
    top: 2rem;
    left: 50%;
    transform: translateX(-50%);
    padding: 1rem;
    border-style: outset;
    border-radius: 9px;
  }
}


