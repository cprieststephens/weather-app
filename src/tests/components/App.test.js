import { render, waitFor } from "@testing-library/react";
import axios from "axios";
import App from "../../components/App";

describe("App", () => {
  it("renders correctly", async () => {
    const mockData = {
      data: {
        location: {
          city: "Stub city",
          country: "Stub country",
        },
        forecasts: [
          {
            date: 11111111,
            temperature: {
              max: 10,
              min: 2,
            },
            wind: {
              speed: 5,
              direction: "n",
            },
            humidity: 20,
            description: "Stub description",
            icon: 800,
          },
        ],
      },
    };
    jest.spyOn(axios, "get").mockResolvedValue(mockData);
    const { asFragment } = render(<App />);
    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
