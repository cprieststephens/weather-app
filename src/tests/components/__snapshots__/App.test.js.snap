// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <div
      class="weather-app__top"
    >
      <div
        class="search-form"
      >
        <input
          placeholder="Manchester"
          type="search"
          value=""
        />
        <button
          type="submit"
        >
          Search
        </button>
      </div>
      <h1>
        , 
      </h1>
    </div>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
